stdlib/Test/src/logging.jl

```diff
@@ -285,12 +306,32 @@ logfield_contains(a, r::Regex) = occursin(r, a)
 logfield_contains(a::Symbol, r::Regex) = occursin(r, String(a))
 logfield_contains(a::LogLevel, b::Symbol) = a == parse_level(b)
 logfield_contains(a, b::Ignored) = true
+logfield_contains(a::NamedTuple, b) = logfield_contains(pairs(a), b)
```

Comment:
I was expecting to see a `Dict` overload here as well; is it somehow covered by other methods? I see you have a test for passing `kwargs` as a Dict, so I also assume it's working somehow?

Reply:
👍 yeah, note that i'm matching this on `a`, not `b`, which is the _LogRecord field_, not the user-supplied Pattern.

I did that because i wanted to be more generic in what types i accepted in the Pattern, and the LogRecord will always have a fixed type field.

Does that seem reasonable to you as well?

🤔 Although, now that i think about it, i guess this could cause method ambiguities? Like if you try to use `Ingored` on the kwargs, i think it would give a method ambiguity. Maybe that's the only one i need to worry about though. Maybe i should just add an overload just for that case? I'll do that now.

Reply:
K, done: i added `logfield_contains(a::NamedTuple, b::Ignored) = true  # method amibguity resolution`

Reply:
:/ this is still causing a method ambiguities test failure though:

```julia
Error in testset ambiguous:
Test Failed at /cache/build/default-amdci5-4/julialang/julia-master/julia-5647ec6a75/share/julia/test/ambiguous.jl:187
  Expression: isempty(detect_ambiguities(modules...; recursive = true, allowed_undefineds))
   Evaluated: isempty(Tuple{Method, Method}[(logfield_contains(a, b::Test.Ignored) @ Test /cache/build/default-amdci5-4/julialang/julia-master/julia-5647ec6a75/share/julia/stdlib/v1.9/Test/src/logging.jl:308, logfield_contains(a::Base.Pairs, pattern) @ Test /cache/build/default-amdci5-4/julialang/julia-master/julia-5647ec6a75/share/julia/stdlib/v1.9/Test/src/logging.jl:311), (logfield_contains(a, r::Regex) @ Test /cache/build/default-amdci5-4/julialang/julia-master/julia-5647ec6a75/share/julia/stdlib/v1.9/Test/src/logging.jl:305, logfield_contains(a::Base.Pairs, pattern) @ Test /cache/build/default-amdci5-4/julialang/julia-master/julia-5647ec6a75/share/julia/stdlib/v1.9/Test/src/logging.jl:311), (logfield_contains(a, r::Regex) @ Test /cache/build/default-amdci5-4/julialang/julia-master/julia-5647ec6a75/share/julia/stdlib/v1.9/Test/src/logging.jl:305, logfield_contains(a::NamedTuple, b) @ Test /cache/build/default-amdci5-4/julialang/julia-master/julia-5647ec6a75/share/julia/stdlib/v1.9/Test/src/logging.jl:309)])
```
It's unhappy about this:
```julia
[logfield_contains(a, b::Test.Ignored),
 logfield_contains(a::Base.Pairs, pattern),
 logfield_contains(a, r::Regex),
 logfield_contains(a::Base.Pairs, pattern),
 logfield_contains(a, r::Regex),
 logfield_contains(a::NamedTuple, b)]
```
so i guess i'll probably need to resolve the ambiguities after all